src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/EditDistance.cs

```diff
@@ -58,18 +58,17 @@ private static char[] ConvertToLowercaseArray(string text)
         {
             var array = ArrayPool<char>.GetArray(text.Length);
             for (var i = 0; i < text.Length; i++)
-            {
                 array[i] = CaseInsensitiveComparison.ToLower(text[i]);
-            }
 
             return array;
         }
 
         public void Dispose()
         {
+            if (_sourceLowerCaseCharacters == null)
+                throw new ObjectDisposedException(nameof(EditDistance));
```

Comment:
📝 I don't think this is working the way one would expect. This would only throw for the case of `default(EditDistance)`.

Reply:
you are correct.  will fix.